app/backend/api.ts

```diff
@@ -35,13 +35,14 @@ export const getPaymentData = async (searchString: string, offset: number, filte
 	})
 
 	if(!response) {
-		return { results: [], total: 0 }
+		return { results: [], total: 0, filterOptions: {} }
```

Comment:
I think you still need to pass the filteroptions back. Even though there are no response for the the given query, dont you want to restore the existing filter options?

Reply:
That is right, if no filterOptions are returned from backend calling function the results page displays all filters.

Reply:
ok